core/backend/src/CheckpointManager.ts

```diff
@@ -270,6 +270,13 @@ export class V1CheckpointManager {
         const dbChangeSetId = nativeDb.getParentChangeSetId();
         if (dbChangeSetId !== checkpoint.mergedChangeSetId)
           throw new IModelError(IModelStatus.ValidationFailed, "ParentChangeSetId of the checkpoint was not correctly setup", Logger.logError, loggerCategory, () => ({ ...traceInfo, ...checkpoint, dbChangeSetId }));
+
+        const dbGuid = Guid.normalize(nativeDb.getDbGuid());
+        if (dbGuid !== Guid.normalize(requestedCkp.iModelId)) {
+          Logger.logWarning(loggerCategory, "iModelId is not properly setup in the briefcase. Updated briefcase to the correct iModelId.", () => ({ ...traceInfo, ...checkpoint, dbGuid }));
+          nativeDb.setDbGuid(requestedCkp.iModelId);

```

Comment:
should you normalize before storing?

Reply:
does this fix your test?

Reply:
Just pushed the change to normalize, thanks.

I tested it locally and it's properly setting the Guid and returning it to the frontend for future requests.  However, the iModel in the Hub will continue to have the incorrect DbGuid since we do not push this change back to iModelHub so every time that briefcase is downloaded it'll need to be updated.

I'll reach out to the Hub team to see what we want to do about this.